cmd/quadlet/main.go

```diff
@@ -103,28 +103,58 @@ func Debugf(format string, a ...interface{}) {
 func getUnitDirs(rootless bool) []string {
 	// Allow overriding source dir, this is mainly for the CI tests
 	unitDirsEnv := os.Getenv("QUADLET_UNIT_DIRS")
+	dirs := make([]string, 0)
+
 	if len(unitDirsEnv) > 0 {
-		return strings.Split(unitDirsEnv, ":")
+		if !isValidFilePath(unitDirsEnv) {
+			Logf("%s not a valid file path", unitDirsEnv)
+			return nil
+		}
+		for _, eachUnitDir := range strings.Split(unitDirsEnv, ":") {
+			dirs = appendSubPaths(dirs, eachUnitDir)
+		}
+		return dirs
 	}
 
-	dirs := make([]string, 0)
 	if rootless {
 		configDir, err := os.UserConfigDir()
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "Warning: %v", err)
 			return nil
 		}
-		dirs = append(dirs, path.Join(configDir, "containers/systemd"))
+		dirs = appendSubPaths(dirs, path.Join(configDir, "containers/systemd"))
 		u, err := user.Current()
 		if err == nil {
-			dirs = append(dirs, filepath.Join(quadlet.UnitDirAdmin, "users", u.Uid))
+			dirs = appendSubPaths(dirs, filepath.Join(quadlet.UnitDirAdmin, "users", u.Uid))
 		} else {
 			fmt.Fprintf(os.Stderr, "Warning: %v", err)
 		}
 		return append(dirs, filepath.Join(quadlet.UnitDirAdmin, "users"))
 	}
 	dirs = append(dirs, quadlet.UnitDirAdmin)
```

Comment:
Why don't you do the same for here?

Reply:
From what I understood, recursive runs were not required in system quadlets. I was thinking `UnitDirAdmin` would be where admin unit files would go, hence system quadlets be run.
https://github.com/containers/podman/pull/18407#issuecomment-1533507261